exporter/carbonexporter/exporter.go

```diff
@@ -137,20 +139,22 @@ func (cp *connPool) Write(bytes []byte) (int, error) {
 	return n, err
 }
 
-func (cp *connPool) Close() {
+func (cp *connPool) Close() error {
 	cp.mtx.Lock()
 	defer cp.mtx.Unlock()
 
+	var errs error
 	for _, conn := range cp.conns {
-		conn.Close()
+		errs = multierr.Append(errs, conn.Close())
```

Comment:
can we use the `errors.Join` method here?

Reply:
I can if I really want, but is not as performant as append for this use-case. It replaces nicely `multierr.Combine` but not the append.

Reply:
We have https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/25121 to track switching to `errors.Join`.

Reply:
So will do in a separate PR then when we solve that.